File: src/test/java/com/github/liebharc/queryenrichment/Attributes.java
package com.github.liebharc.queryenrichment;

public class Attributes {

    public static final Attribute<Long> studentId = new Attribute<>(Long.class,"student", "id");
    public static final Attribute<String> firstName = new Attribute<>(String.class, "student", "firstName");
    public static final Attribute<String> lastName = new Attribute<>(String.class, "student", "lastName");
    public static final Attribute<Long> studentClass = new Attribute<>(Long.class, "student", "class");
    public static final Attribute<String> fullName = new Attribute<>(String.class, "student", "fullName");
    public static final Attribute<String> classIdString = new Attribute<>(String.class, "student", "idString");


    public static final Attribute<Long> teacherId = new Attribute<>(Long.class, "teacher", "id");
    public static final Attribute<String> teacherFirstName = new Attribute<>(String.class, "teacher", "firstName");
    public static final Attribute<String> teacherLastName = new Attribute<>(String.class, "teacher", "lastName");
    public static final Attribute<Long> teacherClass = new Attribute<>(Long.class, "teacher", "class");

    public static final Attribute<Long> classId = new Attribute<>(Long.class, "class", "id");
    public static final Attribute<String> classDescription = new Attribute<>(String.class, "class", "description");

    private Attributes() {

    }
}

File: src/test/java/com/github/liebharc/queryenrichment/Attributes.kt
package com.github.liebharc.queryenrichment

object Attributes {

    val studentId = Attribute(Long::class.java, "student", "id")
    val firstName = Attribute(String::class.java, "student", "firstName")
    val lastName = Attribute(String::class.java, "student", "lastName")
    val reference = Attribute(Student::class.java, "student", Attribute.reference)
    val studentClass = Attribute(Long::class.java, "student", "class")
    val fullName = Attribute(String::class.java, "student", "fullName")
    val classIdString = Attribute(String::class.java, "student", "idString")


    val teacherId = Attribute(Long::class.java, "teacher", "id")
    val teacherFirstName = Attribute(String::class.java, "teacher", "firstName")
    val teacherLastName = Attribute(String::class.java, "teacher", "lastName")
    val teacherClass = Attribute(Long::class.java, "teacher", "class")

    val classId = Attribute(Long::class.java, "class", "id")
    val classDescription = Attribute(String::class.java, "class", "description")
}
